arroyo/processing/strategies/produce.py

```diff
@@ -77,6 +77,7 @@ def poll(self) -> None:
                             offset=message.offset,
                             headers=message.payload.headers,
                             key=message.payload.key,
+                            reason=str(exc),
```

Comment:
For reason I mean pass the original exception:

```
raise WhateverException from exc
```

Still I think this should not be an InvalidMessage

Reply:
Aren't all exceptions from strategies supposed to be InvalidMessage? DLQ needs to work with every strategy and that will not be possible if not.